### ISSUE TITLE
Exception: cell type inlineStr has unexpected child <{http://schemas.openxmlformats.org/spreadsheetml/2006/main}f> at rowx=37 colx=2

### ISSUE THREAD

**Issue by SheetJSDev** [python-excel/xlrd#81]
Sample File: http://oss.sheetjs.com/test_files/LONumbers.xlsx

```
$ runxlrd.py 2rows test_files/LONumbers.xlsx

=== File: test_files/LONumbers.xlsx ===
*** Open failed: Exception: cell type inlineStr has unexpected child <{http://schemas.openxmlformats.org/spreadsheetml/2006/main}f> at rowx=37 colx=2
Traceback (most recent call last):
  File "/usr/local/bin/runxlrd.py", line 333, in main
    ragged_rows=options.ragged_rows,
  File "/Library/Python/2.7/site-packages/xlrd/__init__.py", line 416, in open_workbook
    ragged_rows=ragged_rows,
  File "/Library/Python/2.7/site-packages/xlrd/xlsx.py", line 754, in open_workbook_2007_xml
    x12sheet.process_stream(zflo, heading)
  File "/Library/Python/2.7/site-packages/xlrd/xlsx.py", line 520, in own_process_stream
    self_do_row(elem)
  File "/Library/Python/2.7/site-packages/xlrd/xlsx.py", line 675, in do_row
    bad_child_tag(child_tag)
  File "/Library/Python/2.7/site-packages/xlrd/xlsx.py", line 538, in bad_child_tag
    raise Exception('cell type %s has unexpected child <%s> at rowx=%r colx=%r' % (cell_type, child_tag, rowx, colx))
Exception: cell type inlineStr has unexpected child <{http://schemas.openxmlformats.org/spreadsheetml/2006/main}f> at rowx=37 colx=2
```

**Comment by sylvainr** [comment]
I am still having the original problem with a spreadsheet of mine, generated using LibreOffice 4.2.7.2. Exact same problem when I try to open the LONumbers.xlsx file referenced in the original post.

I am using xlrd 0.9.3.

Because I rely on xlrd and libreoffice (and have to use xlsx files) for an important piece of my workflow, I had to comment the following line (see arrow):

```
elif cell_type == "inlineStr":
    # Not expected in files produced by Excel.
    # Only possible child is <is>.
    # It's a way of allowing 3rd party s/w to write text (including rich text) cells
    # without having to build a shared string table
    for child in cell_elem:
        child_tag = child.tag
        if child_tag == IS_TAG:
            tvalue = get_text_from_si_or_is(self, child)
        else:
======>     #bad_child_tag(child_tag)
            pass
    assert tvalue is not None
    self.sheet.put_cell(rowx, colx, XL_CELL_TEXT, tvalue, xf_index)
```

The text in that block sort of mentions that Excel should not generate such tags. I am using LibreOffice and it may be the reason?

Is just commenting out the raise of the exception going to bite me back later on?

Is it something we should change in xlrd in order to tolerate LibreOffice/OpenOffice generated files?

**Comment by jesusVMayor** [comment]
Same problem with a file with formulas.
If i import a file created with microsoft excel in formats xlsx or xls or with libreoffice in format xls i have no problem.
But if i import a file created or edited with libreoffice in xlsx format fails in the column with the first formula.
Exception: cell type inlineStr has unexpected child <{http://schemas.openxmlformats.org/spreadsheetml/2006/main}f> at rowx=10 colx=128
xlrd version: 0.9.4
libreoffice version: 4.2.8.2

Edit: updated to version 5.1.2.2 of libreoffice the problem disappears

**Comment by cjw296** [comment]
Should be fixed in 1.0.1
